workflows/sweep_simulate.snake

```diff
@@ -0,0 +1,432 @@
+    resources: time=3000, mem_mb=4000
+    run:
+        model = species.get_demographic_model(demo_model["id"])
+        mutation_rate = model.mutation_rate
+        extended_events = None
+        # Use new sample specification, give selected pop and sample size as config arguments
+        samples = demo_model["samples"]
+
+        # Extract contig spanning window of chromosome
+        window_left = int(wildcards.left)
+        window_right = int(wildcards.right)
+        b_window_left, b_window_right = make_buffer(recmap, config["buffer_cM"], window_left, window_right)
+        contig = species.get_contig(chrom, genetic_map=genetic_map_id, left=b_window_left, right=b_window_right)
+        if wildcards.dfe != "NA":
+            dfe = species.get_dfe(wildcards.dfe)
+
+            ## Adding annotation only seletion on exon region
+            annot = species.get_annotations(wildcards.annot)
+            annot_intervals = annot.get_chromosome_annotations(chrom)
+            contig.add_dfe(intervals=annot_intervals, DFE=dfe)
+
+        # sweep parameters
+        coord = ((window_left+window_right)//2) # the API does the right thing even if it is chunked
+        coeff = float(wildcards.coeff)
+        # Defining how long ago beneficial mutations arise
+        # First getting 2N from the model and pop
+        debug = model.model.debug()
+        coal_T = debug.mean_coalescence_time({wildcards.popu: 2})
+        gamma = coal_T* coeff #2Ns
+        T_f = 4 * (np.log(gamma) + 0.5772 - (1/gamma)) / coeff # this is from Charlesworth 2020/Pennings and Hermisson 2007
+        time = np.floor(1.5*T_f) # I dilated the time a bit because of variance (don't want to lose too many sims due to this -- note we are conditioning on sweep_min_freq)
+        print(coal_T, gamma, T_f, time)
+        contig.add_single_site("sweep", coordinate=coord)
+        extended_events = stdpopsim.ext.selective_sweep(
+            "sweep",
+            population=wildcards.popu, #selected pop
+            selection_coeff=coeff,
+            mutation_generation_ago=time,
+            min_freq_at_end=config["sweep_min_freq"]
+        )
+        
+        contig.mutation_rate = mutation_rate
+        engine = stdpopsim.get_engine("slim")
+        ts = engine.simulate(
+            model,
+            contig,
+            samples,
+            extended_events=extended_events,
+            slim_scaling_factor=config["slim_scaling_factor"],
+            slim_burn_in=config["slim_burn_in"],
+            seed=int(wildcards.seed),
+        )
+        ts = add_window_metadata(ts, window_left, window_right, b_window_left, b_window_right, extra=f"sweep_time={time};")
+        ts.dump(output[0])
+
+def _get_stats(input, output, params_dict, target_pops):
+    meta = ["model","demo","seed","chrom","left","right","annot","dfe","coeff"]
+    stats = ["taj"]
```

Comment:
would be good to get a windowed look at \pi in the focal window just to confirm that we are parameterizing sims in a way that allows us to see a hitchhiking effect

Reply:
in looking at mean TajD over a small number of replicates it looks like we may need to tune the sweep params a bit 
![image](https://user-images.githubusercontent.com/7674197/214374853-dde2456a-56a2-4421-9d71-2d977cc3e790.png)